onnx/shape_inference/implementation.cc

```diff
@@ -426,7 +426,9 @@ static void InferShapesImpl(
       }
       for (int i = 0; i < n.output_size(); ++i) {
         auto* inferredType = ctx.getOutputType(i);
-        if (inferredType->value_case() == TypeProto::ValueCase::VALUE_NOT_SET) {
+        // If an output is missing (lacks name), shape inference should not infer shape
+        if (inferredType->value_case() == TypeProto::ValueCase::VALUE_NOT_SET ||
+          n.output(i).empty()) {
```

Comment:
suggest making this a separate condition and adding it above line 428. We should not even fetch the inferredType with output is empty.

On a separate note why is there no check for inferredType != nullptr? Can we safely assume ctx.getOutputType(i) never returns a nullptr?

Reply:
> suggest making this a separate condition and adding it above line 428. We should not even fetch the inferredType with output is empty.

Sounds good and I just moved the condition 

> On a separate note why is there no check for inferredType != nullptr? Can we safely assume ctx.getOutputType(i) never returns a nullptr?

For now I think it should be fine because the vector of output type will be initialized here: https://github.com/onnx/onnx/blob/master/onnx/shape_inference/implementation.h#L184